src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -4104,38 +4144,70 @@ WorkerLimitCount(MultiExtendedOp *originalOpNode)
 	 */
 	if (groupClauseList == NIL)
 	{
-		workerLimitCount = originalOpNode->limitCount;
+		canPushDownLimit = true;
 	}
-	else if (sortClauseList != NIL)
+	else if (sortClauseList == NIL)
 	{
-		bool orderByNonAggregates = !(HasOrderByAggregate(sortClauseList, targetList));
-		bool canApproximate = CanPushDownLimitApproximate(sortClauseList, targetList);
-
-		if (orderByNonAggregates)
+		if (!hasAggregateTarget)
```

Comment:
This is the only place which we use hasAggregateTarget. We can move declaration to here.